apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/OperationExtensions.kt

```diff
@@ -177,3 +179,32 @@ fun <D : Operation.Data, M: Map<String, Any?>> Operation<D>.parseData(
     adapter(responseAdapterCache).fromResponse(it)
   }
 }
+
+fun <D : Operation.Data> Operation<D>.variables(responseAdapterCache: ResponseAdapterCache): Operation.Variables {
```

Comment:
hm looks like this API is expensive, as we always do serialization for variables. Also how user can easily get access to variables for the instance of the query? With this API it needs to provide some internal `ResponseAdapterCache `